### PR TITLE
Fix problems found by SonarLint

### PR DESCRIPTION
**Pull request by AkosUzonyi**

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
   private int torpedoCount = 0;
 
+  public Random generator = new Random();
```
Comment (ghost) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: This random generator has been used locally only. This should be private.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+    throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (ghost) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: You should add tab before `throw` for a more uniform code.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -=- numberOfTorpedos;
```
Comment (ghost) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: You probably wanted to write `-= n`, but you wrote `-= -n` instead.

**Review by ghost — CHANGES_REQUESTED** [review]
I've found two serious bugs and a formatting fault. You may fix all of these problems.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (AkosUzonyi) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Okay, I'll make it private

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 2 above)
Comment (AkosUzonyi) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Wow, good catch. Thanks, will be fixed :).

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 3 above)
Comment (AkosUzonyi) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Oh, SonarLint was finally stopped complaining after this change, so I thought it was ok now. But agreed, let's write `-=`, that looks simpler :D.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
   private int torpedoCount = 0;
 
-  public Random generator = new Random();
+  private Random generator = new Random();
```
Comment (ghost) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: Correct fix.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 5]
```diff
@@ -32,7 +32,7 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-    throw new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (ghost) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: Thanks for sticking to our formatting convention.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 6]
```diff
@@ -42,7 +42,7 @@ public boolean fire(int numberOfTorpedos){
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount -=- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (ghost) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: This solution finally seems logical. I'll accept it.

**Review by ghost — APPROVED** [review]
All bugs have been fixed.

**Pull request merged**